clusterDeployer.py

```diff
@@ -33,6 +33,149 @@ def match_to_proper_version_format(version_cluster_config: str) -> str:
+        return
+
+    content_rc = b""
+    if not os.path.exists(rc_orig):
+        try:
+            with open(rc, "rb") as f:
+                content_rc = f.read()
+        except IOError:
+            pass
+        # Backup the original resolv.conf file to "rc_orig"
+        if content_rc == b"":
+            pass
+        elif b"Written by cluster-deployment-automation" in content_rc:
+            content_rc = b""
+        elif b"Generated by NetworkManager" in content_rc and os.path.exists("/run/NetworkManager/resolv.conf"):
+            # This file seems to be written by NetworkManager. Symlink there.
+            logger.info(f"resolv.conf: point {rc_orig} to /run/NetworkManager/resolv.conf")
+            lh.run_or_die(f"ln -snf /run/NetworkManager/resolv.conf {rc_orig}")
+        else:
+            logger.info(f"resolv.conf: write {rc_orig}")
+            with common.atomic_write(rc_orig, text=False) as f:
+                f.write(content_rc)
+    else:
+        try:
+            with open(rc_orig, "rb") as f:
+                content_rc = f.read()
+        except IOError:
+            pass
+
+    with common.atomic_write(rc_local) as file:
+        file.write(
+            f"""# Written by cluster-deployment-automation.
+#
+# A local dnsmasq instance is configured (`systemctl status dnsmasq.service`).
+# Original file is at "{rc_orig}".
+#
+# Set:     `sudo ln -snf {os.path.basename(rc_local)} /etc/resolv.conf`
+# Restore: `sudo ln -snf {os.path.basename(rc_orig)} /etc/resolv.conf`
+# Alternatively, restore by deleting "/etc/resolv.conf" and let NetworkManager
+# write anew with `systemctl reload NetworkManager.service`.
+#
+# Prevent touching "/etc/resolv.conf" with a file "/etc/.resolv.conf.cda-dont-touch"
+
+nameserver 127.0.0.1
+"""
+        )
+    # We only point a symlink to the correct file. This indicates that
+    # somebody is handling the file and NetworkManager leaves the symlink
+    # alone.
+    # It's also easy to restore the previous file with
+    # `ln -snf resolv.conf.cda-orig /etc/resolv.conf`.
+    if not dont_touch:
+        lh.run_or_die(f"ln -snf {os.path.basename(rc_local)} {rc}")
+
+
+def _update_dnsmasq(cluster_name: str, api_vip: Optional[str] = None) -> None:
+
+    lh = host.LocalHost()
+
+    service_running = lh.run("systemctl is-active --quiet dnsmasq.service").success()
+
+    service_restart = True
+
+    dmasqconf_my = f"/etc/dnsmasq.d/cda-{cluster_name}.conf"
+
+    if api_vip is None:
+        loglevel = logging.INFO
+        try:
+            os.remove(dmasqconf_my)
+        except FileNotFoundError:
+            loglevel = logging.DEBUG
+            service_restart = False
+        except OSError:
+            service_restart = service_running
+        logger.log(loglevel, f"dnsmasq: reset local dnsmasq by deleting {dmasqconf_my}")
+    else:
+        content = f"# Written by cluster-deployment-automation for \"{cluster_name}\"\nserver=/{cluster_name}.redhat.com/{api_vip}\n"
```

Comment:
Don't we need to add the default servers too?  I.e. the ones we had in `/etc/resolv.conf`?

Otherwise we end up with a dnsmasq config that looks like:
```
# cat /etc/dnsmasq.d/cda-ovnscale.conf 
# Written by cluster-deployment-automation for "ovnscale"
server=/ovnscale.redhat.com/192.168.124.250
```
And no rules to resolve anything not included in `*.ovnscale.redhat.com`.

Reply:
right... will work on solution. Thank you!!

Reply:
should work now...

Reply:
It's working now, thanks!